### PR TITLE
build: Use go cross compiling instead of qemu

### PR DESCRIPTION
**Pull request by qinqon**
**Is this a BUG FIX or a FEATURE ?**:
The buildx build command start a QEMU instance if the container arch is
not the one on the host, go toolchain has cross compiling functionality
so qemu is not needed for the compiling stuff. This change force the
compile stage to be amd64 since it's the arch at CI and then runs cross
compiling to compile amd64 or arm64 so it does not compile over qemu.

> Uncomment only one, leave it on its own line:
>
> /kind bug
> /kind enhancement

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -2,11 +2,15 @@ FROM --platform=linux/amd64 quay.io/centos/centos:stream8 AS build
...
-RUN dnf install -y golang-1.17.*
+RUN mkdir /workdir
 
-COPY . /workdir
+COPY go.mod /workdir
 WORKDIR /workdir
 
+RUN dnf install -y golang-$(sed -En 's/^go +(.*+)$/\1/p' go.mod).*
+
+COPY . .
```
Comment (rhrazdil) [inline build/Dockerfile:12]: What is this magic?

Comment (qinqon) [reply build/Dockerfile:12]: We have already change to /workdir so it's not needed to be specified

Comment (rhrazdil) [reply build/Dockerfile:12]: Ah of course

**Review by rhrazdil — CHANGES_REQUESTED** [review]
Could you please update also the build/Dockerfile.future?
Or do you wish to merge this one and update future in a followup?

**Comment by rhrazdil** [comment]
/lgtm
Thank you!

**Pull request merged**